identity/src/peer_id.rs

```diff
@@ -137,12 +123,13 @@ impl PeerId {
     /// Returns `None` if this `PeerId`s hash algorithm is not supported when encoding the
     /// given public key, otherwise `Some` boolean as the result of an equality check.
     pub fn is_public_key(&self, public_key: &crate::PublicKey) -> Option<bool> {
-        use multihash::MultihashDigest as _;
+        let other_peer_id = PeerId::from_public_key(public_key);
 
-        let alg = Code::try_from(self.multihash.code())
-            .expect("Internal multihash is always a valid `Code`");
-        let enc = public_key.encode_protobuf();
-        Some(alg.digest(&enc) == self.multihash)
+        if self.multihash.code() != other_peer_id.multihash.code() {
+            return None;
+        }
```

Comment:
I thought the same when I was changing this code. What this currently supports is a possible extension of the `PeerId` spec where we encode public keys with a different hash function. i.e. say we specify tomorrow that we also can hash public keys with SHA3. For a SHA3-hashed `PeerId`, this will return `None`. If we just compare the multihashes, we would return `Some(false)` or just `false` if we remove the `Option`.

This `false` could be a false-positive if it is actually the same public key, just encoded with a different hash function.

We don't use this function anywhere so I think we can just remove this. The functionality can be replicated outside of this crate if somebody really needs it because we expose the `Multihash`.